src/main/java/org/opensearch/security/dlic/rest/api/ssl/CertificatesNodesResponse.java

```diff
@@ -0,0 +1,157 @@
+                builder.startObject("load_exception");
+                OpenSearchException.generateThrowableXContent(builder, params, node.exception);
+                builder.endObject();
+            }
+            if (node.certificates() != null) {
+                for (final var e : node.certificates().entrySet()) {
+                    builder.field(e.getKey(), e.getValue());
+                }
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static class NodeResponse extends BaseNodeResponse {
+
+        private final Exception exception;
+
+        private final Map<String, Map<String, List<Map<String, String>>>> certificates;
+
+        public NodeResponse(final DiscoveryNode node, final Exception exception) {
+            super(node);
+            this.exception = exception;
+            this.certificates = null;
+        }
+
+        public NodeResponse(final DiscoveryNode node, final Map<String, Map<String, List<Map<String, String>>>> certificates) {
+            super(node);
+            this.exception = null;
+            this.certificates = certificates;
+        }
+
+        public NodeResponse(StreamInput in) throws IOException {
+            super(in);
+            if (in.readBoolean()) {
+                this.exception = in.readException();
+                this.certificates = null;
+            } else {
+                this.exception = null;
+                this.certificates = in.readMap(
+                    StreamInput::readString,
+                    mapIn -> mapIn.readMap(
+                        StreamInput::readString,
+                        listIn -> listIn.readList(min -> min.readMap(StreamInput::readString, StreamInput::readOptionalString))
+                    )
+                );
+            }
+        }
+
+        public Map<String, Map<String, List<Map<String, String>>>> certificates() {
+            return certificates;
+        }
+
+        public Exception exception() {
+            return exception;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            if (exception != null) {
+                out.writeBoolean(true);
+                out.writeException(exception);
+            }
+            if (certificates != null) {
+                out.writeBoolean(false);
+                out.writeMap(
+                    certificates,
+                    StreamOutput::writeString,
+                    (mapOut, map) -> mapOut.writeMap(
+                        map,
+                        StreamOutput::writeString,
+                        (listOut, list) -> listOut.writeCollection(
+                            list,
+                            (mo, m) -> mo.writeMap(m, StreamOutput::writeString, StreamOutput::writeOptionalString)
+                        )
+                    )
+                );
```

Comment:
the nested writeMap is slightly confusing to read. Mind adding an example of how it would look post this code block is executed?